src/figuresClasses.ts

```diff
@@ -2,10 +1,81 @@
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.color = color;
+
+    const ab = a + b;
+    const bc = b + c;
+    const ac = a + c;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('any length is <= 0');
+    }
+
+    if (a >= bc || b >= ac || c >= ab) {
+      throw new Error('it is not a triangle');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public r: number,
+  ) {
+    this.color = color;
+
+    if (r <= 0) {
+      throw new Error('this is a dot');
+    }
+  }
 
-export class Triangle implements Figure {}
+  getArea(): number {
+    const area = Math.PI * Math.pow(this.r, 2);
 
-export class Circle implements Figure {}
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'rectangle' = 'rectangle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+  ) {
+    if (a <= 0 || b <= 0) {
+      throw new Error('any length is <= 0');
+    }
+    this.color = color;
+  }
+
+  getArea(): number {
+    return this.a * this.b;
+  }
+}
 
 export function getInfo(figure): string {
```

Comment:
The `getInfo` function should specify the type of its parameter. Consider changing the parameter to `figure: Figure` to ensure type safety.